### ISSUE TITLE
Output a list of all scenarios

### ISSUE THREAD

**Issue by zrno** [catchorg/Catch2#2250]
**Description**
I would like to output all the scenarios into a file, no matter the success or failure. I would like to use this output to give it to the users of my code so that they can evaluate all the scenarios and provide feedback if any scenario was missed.

The output should include all texts in all the SCENARIO, GIVEN, WHEN, THEN descriptions nicely formatted so that even non-programmers could read it.

```
Scenario: Something is inside
      Given: Somethins is inside something else
  And given: Something is not green
       When: Something happens
   And when: Something else happens
       Then: Disaster

Scenario: Something is outside
      Given: Somethins is outside something else
  And given: Something is not green
       When: Something happens
   And when: Something else happens
       Then: Disaster

.....
```

**Comment by horenmar** [comment]
Make a custom reporter, hook test case and section ending events and write them out however you want. (They should be well formatted by default, but you can change the formatting if you want)

**Comment by zrno** [comment]
Thx. Would that give duplicated outputs if GENERATOR is used in the test cases?

**Comment by horenmar** [comment]
Yes. For all intents and purposes, generators multiply the test case. If you want to disambiguate between the test case runs with different generated data you should `CAPTURE` the data and also incorporate these in your output.

**Comment by zrno** [comment]
I see. But I don't want that.
Merely outputting all the strings in the all the macros once would be enough.
I can write a small parser that does it, but that is not the point. 

I feel a simple command line flag would be great and the output would be very useful
